site/content/languages/zig.md

```diff
@@ -0,0 +1,213 @@
+below `main` function into a WASI function exported as `_start`.
+
+```zig
+const std = @import("std");
+const io = std.io;
+const os = std.os;
+const allocator = std.heap.page_allocator;
+const stdout = std.io.getStdOut();
+```
+
+Note: wazero includes an [example WASI project][10] including [source code][11]
+that implements `cat` without any WebAssembly-specific code.
+
+## Concurrency
+
+Please read our overview of WebAssembly and
+[concurrency]({{< ref "_index.md#concurrency" >}}). In short, the current
+WebAssembly specification does not support parallel processing.
+
+## Optimizations
+
+Below are some commonly used configurations that allow optimizing for size or
+performance vs defaults. Note that sometimes one sacrifices the other.
+
+### Binary size
+
+Those with `%.wasm` binary size constraints can change their source, 
+e.g. picking a [different allocator][9b] or set `zig` flags to reduce it.
+
+Source changes:
+* [WebAssembly-only fast allocator][12] is a specialized allocator that 
+trades generality for simplicity, making however specific assumptions 
+on the runtime environment (e.g. single-threaded access to global variables, 
+it direcly calls Wasm-specific intrinsics and it never releases memory)
+
+[`zig` flags][13]:
+Zig provides several flags to control binary size, speed of execution, 
+safety checks. For instance you may use
+* `-ODebug`: Fast build, enabled safety checks, slower runtime performance, 
+  larger binary size
+* `-OReleaseSafe`: Medium runtime performance, enabled safety checks, 
+  slower compilation speed, larger binary size
+* `-OReleaseSmall`: Medium runtime performance, disabled safety checks, 
+  slower compilation speed, smaller binary size
+
+### Performance
+
+Those with runtime performance constraints can change their source or set
+`zig` flags to improve it.
+
+[`zig` flags][13]:
+* `-OReleaseFast`: Enable additional optimizations, possibly at the cost of 
+  increased binary size.
+
+## Frequently Asked Questions
+
+### Why is my `%.wasm` binary so big?
+Zig defaults can be overridden for those who can sacrifice features or
+performance for a [smaller binary](#binary-size). After that, tuning your
+source code may reduce binary size further.
+
+[1]: https://ziglang.org/download/0.4.0/release-notes.html
+[2]: https://ziglang.org/documentation/master/#WASI
+[5]: https://discord.gg/gxsFFjE
+[6]: https://github.com/tetratelabs/wazero/tree/main/site/content/languages/zig.md
+[7]: https://github.com/tetratelabs/wazero/stargazers
+[9]: https://github.com/tetratelabs/wazero/tree/main/examples/allocation/zig
+[9b]: https://ziglang.org/documentation/0.10.1/#Memory
+[10]: https://github.com/tetratelabs/wazero/tree/main/imports/wasi_snapshot_preview1/example/testdata/zig
+[11]: https://github.com/tetratelabs/wazero/blob/main/imports/wasi_snapshot_preview1/example/testdata/zig/cat.zig
+[12]: https://github.com/ziglang/zig/pull/13513
+[13]: https://ziglang.org/documentation/master/#Build-Mode
```

Comment:
can this be a permalink rather than master?

Reply:
fixing `[2]` too